test/parallel/test-buffer-write.js

```diff
@@ -90,6 +90,8 @@ for (let i = 1; i < 4; i++) {
 const z = Buffer.alloc(4, 0);
 assert.strictEqual(z.write('\u0001', 3, 'ucs2'), 0);
 assert.strictEqual(Buffer.compare(z, Buffer.alloc(4, 0)), 0);
+// Make sure longer strings are written up to the buffer end.
+assert.strictEqual(z.write('abcd', 2), 2);
```

Comment:
Most other assertions of this type in this file also verify the Buffer's contents afterwards, maybe we could do that here too?

Reply:
@addaleax, thanks. Added assertion to verify the contents of buffer.